Signal/src/ViewControllers/ProfileViewController.m

```diff
@@ -204,14 +244,44 @@ - (void)setHasUnsavedChanges:(BOOL)hasUnsavedChanges
 {
     _hasUnsavedChanges = hasUnsavedChanges;
 
-    if (hasUnsavedChanges) {
-        self.navigationItem.rightBarButtonItem = (self.hasUnsavedChanges
-                ? [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"EDIT_GROUP_UPDATE_BUTTON",
-                                                             @"The title for the 'update group' button.")
-                                                   style:UIBarButtonItemStylePlain
-                                                  target:self
-                                                  action:@selector(updatePressed)]
-                : nil);
+    [self updateNavigationItem];
+}
+
+- (void)updateNavigationItem
+{
+    // The navigation bar is hidden in the registration workflow.
+    if (self.navigationController.navigationBarHidden) {
+        [self.navigationController setNavigationBarHidden:NO animated:YES];
+    }
+
+    // Always display a left item to leave the view without making changes.
+    // This might be a "back", "skip" or "cancel" button depending on the
+    // context.
+    switch (self.profileViewMode) {
+        case ProfileViewMode_AppSettings:
+            self.navigationItem.leftBarButtonItem =
+                [self createOWSBackButtonWithTarget:self selector:@selector(backOrSkipButtonPressed)];
```

Comment:
Okay, I found a better approach on stackoverflow that works pretty well.  It eliminates the "custom" back buttons everywhere except conversation view.  It's a big change, so PTAL @michaelkirk